### ISSUE TITLE
Bug: `summary.json` contains a single window's metrics instead of an aggregate across all walk-forward runs

### ISSUE THREAD

**Issue by MDUYN** [coding-kitties/investing-algorithm-framework#511]
Bug: `summary.json` contains a single window's metrics instead of an aggregate across all walk-forward runs

## Summary

`backtest_summary` (written to `summary.json`) is supposed to be the **aggregate**
of the per-run `BacktestMetrics` (the `metrics.json` files under each
`runs/<window>/` folder). For multi-window walk-forward backtests it is **not**.
The saved `summary.json` reports `number_of_windows: 1` and contains values that
match exactly **one** of the per-run `metrics.json` files; the other windows are
silently dropped.

This makes every aggregate field in the file wrong (totals, averages, win rate,
profit factor, drawdown, …), not just the loss-related fields originally noticed.
Several secondary bugs in the loss/growth metric definitions are described at the
bottom of this issue, but they are dwarfed by the aggregation bug above.

## Environment

- **Framework**: `investing_algorithm_framework`
- **Version**: 8.6.0
- **Python**: 3.11

## Concrete Example

A walk-forward backtest with 7 yearly windows:

```
backtest_results/.../fff8c54d/
├── summary.json                     ← aggregate (BROKEN)
└── runs/
    ├── backtest_EUR_20221227_20231227/metrics.json   total_net_gain = +499.21, trades = 56, win_rate = 0.393
    ├── backtest_EUR_20230327_20240326/metrics.json   total_net_gain = +467.44, trades = 60, win_rate = 0.400
    ├── backtest_EUR_20230625_20240624/metrics.json   total_net_gain = +408.36, trades = 63, win_rate = 0.365
    ├── backtest_EUR_20230923_20240922/metrics.json   total_net_gain = +376.15, trades = 57, win_rate = 0.404
    ├── backtest_EUR_20231222_20241221/metrics.json   total_net_gain = +222.75, trades = 63, win_rate = 0.333
    ├── backtest_EUR_20240321_20250321/metrics.json   total_net_gain = -110.54, trades = 62, win_rate = 0.226
    └── backtest_EUR_20240619_20250619/metrics.json   total_net_gain =  -24.75, trades = 57, win_rate = 0.316
```

Aggregate sanity checks:
- Sum of `total_net_gain` across windows: **+1838.62**
- Sum of `number_of_trades_closed` across windows: **418**
- Number of windows: **7**

What `summary.json` actually contains:

```json
{
  "total_net_gain": -24.7509,
  "total_net_gain_percentage": -0.02475,
  "number_of_trades_closed": 57,
  "win_rate": 0.3158,
  "number_of_windows": 1,
  "number_of_profitable_windows": 0,
  "number_of_windows_with_trades": 1,
  "average_trade_duration": 361.33,
  "max_consecutive_wins": 6,
  "max_consecutive_losses": 13
}
```

Every value above is identical to the values in
`runs/backtest_EUR_20240619_20250619/metrics.json` (the **last** run only). Six of
seven windows — including five profitable ones — are missing from the aggregate.

## Where the aggregation goes wrong

`Backtest.open(directory_path)` correctly loads all 7 runs from `runs/` and then
calls `generate_backtest_summary_metrics(temp_metrics)` on them
([backtest.py L294-L307](.venv/lib/python3.11/site-packages/investing_algorithm_framework/domain/backtesting/backtest.py#L294-L307)):

```python
if backtest_date_ranges is not None:
    summary_file = os.path.join(directory_path, "summary.json")
    if os.path.isfile(summary_file):
        backtest_summary_metrics = BacktestSummaryMetrics.open(summary_file)
else:
    # Generate new summary from loaded backtest runs
    temp_metrics = []
    for br in backtest_runs:
        if br.backtest_metrics:
            temp_metrics.append(br.backtest_metrics)
    backtest_summary_metrics = generate_backtest_summary_metrics(temp_metrics)
```

That path produces a correct summary at **load** time. The bug is therefore
upstream: the code that **writes** `summary.json` during the original backtest run
is constructing the aggregate from a single window, not the full set. Two suspects:

1. **`Backtest.merge()`** at [backtest.py L506-L521](.venv/lib/python3.11/site-packages/investing_algorithm_framework/domain/backtesting/backtest.py#L506-L521) does:

   ```python
   summary = BacktestSummaryMetrics()
   for bt_run in merged.get_all_backtest_metrics():
       summary.add(bt_run)
   merged.backtest_summary = summary
   ```

   `BacktestSummaryMetrics` is a plain `@dataclass` and **does not define an
   `add()` method** (see
   [backtest_summary_metrics.py](.venv/lib/python3.11/site-packages/investing_algorithm_framework/domain/backtesting/backtest_summary_metrics.py)).
   Calling `summary.add(...)` raises `AttributeError`. If walk-forward results
   are produced by repeated `merge()` calls, the merge silently fails (or is
   wrapped in a `try/except`) and the summary keeps whatever was last assigned.

2. The walk-forward / runner code probably assigns
   `backtest.backtest_summary = generate_backtest_summary_metrics([single_run])`
   per window without ever recomputing it from the **complete** list before
   `Backtest.save()` writes `summary.json`.

Either way, `Backtest.save()` at
[backtest.py L422-L427](.venv/lib/python3.11/site-packages/investing_algorithm_framework/domain/backtesting/backtest.py#L422-L427)
just writes `self.backtest_summary` straight to disk, with no validation that
`number_of_windows` matches `len(self.backtest_runs)`.

## Steps to Reproduce

1. Run a walk-forward / multi-window backtest that produces ≥ 2 runs under
   `runs/`.
2. Compare `summary.json` to the per-run `metrics.json` files:
   ```python
   import json, glob
   runs = [json.load(open(p)) for p in glob.glob('runs/*/metrics.json')]
   summary = json.load(open('summary.json'))
   print("runs:", len(runs))
   print("summary.number_of_windows:", summary['number_of_windows'])
   print("sum(total_net_gain):", sum(r['total_net_gain'] for r in runs))
   print("summary.total_net_gain:", summary['total_net_gain'])
   ```
3. `number_of_windows` is `1`; sums and aggregates do not reconcile.

## Expected Behaviour

- `summary.json` must reflect **all** runs in `runs/`. In particular:
  - `number_of_windows == len(runs)`
  - `number_of_trades_closed == sum(per-run number_of_trades_closed)`
  - `total_net_gain == sum(per-run total_net_gain)`
  - `win_rate` weighted by per-run closed trades
- `Backtest.save()` should regenerate the summary from the current set of runs (or
  validate `number_of_windows`/`number_of_trades_closed` against the runs and
  refuse to write a stale summary).

## Suggested Fixes

1. **Always rebuild the summary at save time** in `Backtest.save()`:

   ```python
   if self.backtest_runs:
       per_run = [br.backtest_metrics for br in self.backtest_runs
                  if br.backtest_metrics is not None]
       self.backtest_summary = generate_backtest_summary_metrics(per_run)
   ```

   This would make the on-disk `summary.json` self-consistent regardless of how
   the in-memory `Backtest` was built up.

2. **Fix or remove `Backtest.merge()`**. Replace `summary.add(bt_run)` with
   `merged.backtest_summary = generate_backtest_summary_metrics(
   merged.get_all_backtest_metrics())`, OR implement
   `BacktestSummaryMetrics.add()` so the dataclass is actually incrementally
   accumulable.

3. **Add a regression test** that runs ≥ 2 windows, saves, reopens, and asserts
   `number_of_windows`, `number_of_trades_closed`, and the sums of net gain /
   gross loss / gross profit reconcile with the per-run values.

---

# Secondary bugs in the loss / growth metric definitions

Even when `summary.json` *does* hold the correct aggregate (e.g. for a single-
window backtest), several individual fields are mathematically inconsistent. These
are independent of the aggregation bug above and apply to **per-run**
`metrics.json` as well as to `summary.json`.

A real per-run `metrics.json` (window `20240619_20250619`):

```json
{
  "total_net_gain": -24.7509,
  "total_net_gain_percentage": -0.02475,
  "total_loss": -24.7509,
  "total_loss_percentage": -0.02475,
  "average_trade_loss": -10.7044,
  "average_trade_loss_percentage": -0.10573,
  "number_of_trades_closed": 57,
  "win_rate": 0.3158
}
```

In the per-run file: `total_loss == total_net_gain` and
`total_loss_percentage == total_net_gain_percentage`.

The single-window `summary.json` derived from that run:

```json
{
  "total_loss": 417.4733,
  "total_loss_percentage": -0.02475,
  "average_loss": 417.4733,
  "average_loss_percentage": -0.02475,
  "average_net_gain": -24.7509,
  "average_growth": -24.7509,
  "total_growth": -24.7509,
  "total_growth_percentage": -0.02475
}
```

In the summary file: `average_loss == total_loss` (€417), every `average_*`
duplicates its `total_*`, and `total_growth*` duplicates `total_net_gain*`.

### B1. Per-run `total_loss` / `total_loss_percentage` is just net return clamped at zero

[`services/metrics/returns.py`](.venv/lib/python3.11/site-packages/investing_algorithm_framework/services/metrics/returns.py#L301)
`get_total_loss`:

```python
def get_total_loss(snapshots):
    ...
    absolute_return = final_value - initial_value
    if absolute_return >= 0:
        return 0.0, 0.0
    percentage = (absolute_return / initial_value)
    return absolute_return, percentage
```

This is not a "total loss" metric. It is `(net_pnl, net_pnl_pct)` clamped to zero
when positive, so per-run `total_loss == total_net_gain` whenever the window is
unprofitable, and `0.0` otherwise. A real "total loss" should be either:
- gross loss (sum of P&L of all losing trades), or
- net realized loss (with an unambiguous, distinct name).

### B2. Aggregate `total_loss` (gross, currency) is paired with `total_loss_percentage` (net %)

In [`combine_backtests.py`](.venv/lib/python3.11/site-packages/investing_algorithm_framework/domain/backtesting/combine_backtests.py#L184-L194):

```python
total_loss = sum(b.gross_loss for b in valid_metrics if b.gross_loss is not None)
total_loss_percentage = _compound_percentage_returns(
    [b.total_loss_percentage for b in valid_metrics]   # net %, see B1
)
```

`total_loss` and `total_loss_percentage` therefore describe different quantities
in different units — gross losses in € vs. compounded net return %.

### B3. `total_growth*` is a duplicate of `total_net_gain*`

`get_total_return` and `get_total_growth` in `returns.py` are byte-for-byte the
same calculation. Either remove `total_growth*` or redefine it (e.g. include
unrealized P&L on still-open positions).

### B4. `average_loss`, `average_net_gain`, `average_growth*` are tautologies for a single window

```python
average_total_loss = safe_weighted_mean(
    [b.gross_loss for b in valid_metrics],
    [b.total_number_of_days for b in valid_metrics]
)
```

For one window, the weighted mean equals the single input — so
`average_loss == total_loss`, `average_net_gain == total_net_gain`, etc. Real
per-trade averages already exist (`average_trade_loss`,
`average_trade_loss_percentage`) and look correct. Suggest renaming these
"average across windows" fields to `*_per_window` (or removing them) and clearly
documenting the weighting basis.

### B5. Mixed-unit percentages fed to `_compound_percentage_returns`

```python
def _compound_percentage_returns(percentages):
    ...
    for pct in valid_percentages:
        compounded *= (1 + pct / 100)        # assumes whole-number %
    return (compounded - 1) * 100             # returns whole-number %
```

But the per-run `*_percentage` fields are decimals (`-0.0247` ≈ −2.47%, not
−0.0247%). This silently produces results off by a factor of ~100 once the
aggregation bug at the top of this issue is fixed and multi-window summaries
actually start exercising this path. Recommendation: make the whole framework
use decimals (`0.10` = 10%) consistently and replace the function with:

```python
def _compound_percentage_returns(percentages):
    valid = [p for p in percentages if p is not None]
    if not valid:
        return None
    compounded = 1.0
    for pct in valid:
        compounded *= (1 + pct)        # decimals
    return compounded - 1
```

## Impact

- Walk-forward / multi-window backtests display headline numbers from a single
  window, hiding the (often very different) performance of the others. Any
  downstream selection / ranking based on `summary.json` is unsound.
- Even when the aggregation is correct (single window), several loss- and growth-
  related fields are duplicates or mismatched in units, leading to misleading
  reporting.
- Once the aggregation bug is fixed, the unit-mismatch bug B5 will start to
  produce additional wrong numbers in multi-window summaries.
